optaplanner-core/src/main/java/org/optaplanner/core/impl/score/buildin/hardsoftdouble/HardSoftDoubleScoreDefinition.java

```diff
@@ -68,6 +69,12 @@ public HardSoftDoubleScore fromLevelNumbers(int initScore, Number[] levelNumbers
         return HardSoftDoubleScore.ofUninitialized(initScore, (Double) levelNumbers[0], (Double) levelNumbers[1]);
     }
 
+    @Override
+    public ScoreInliner<HardSoftDoubleScore> buildScoreInliner(boolean constraintMatchEnabled) {
+        throw new IllegalStateException("ConstraintStreams don't support a " + HardSoftDoubleScore.class.getSimpleName()
```

Comment:
Has `HardSoftDoubleScore` been `@Deprecated`? If so, perhaps it should be.

Reply:
Agreed, nobody uses it, we don't want people to use it - it's just there so people can find it and figure out that they don't want to use it... Deprecated it.

Reply:
Not adding that to this PR, tracking it here for later: https://issues.jboss.org/browse/PLANNER-1579